audio/src/main/java/com/voxeet/audio2/devices/BluetoothDevice.java

```diff
@@ -45,26 +44,24 @@ public BluetoothDevice(
             @NonNull android.bluetooth.BluetoothDevice bluetoothDevice,
             @NonNull __Call<PlatformDeviceConnectionWrapper> wrapper,
             @NonNull __Call<BluetoothDeviceConnectionWrapper> waitForSolver,
-            @NonNull __Call<BluetoothDevice> setActive,
-            @NonNull __Call<BluetoothDevice> onDisconnected) {
+            @NonNull __Call<BluetoothDevice> setActive) {
         super(connectionState, deviceType, bluetoothDevice.getAddress(), bluetoothDevice, bluetoothDevice.getName());
 
         this.bluetoothHeadsetDeviceManager = bluetoothHeadsetDeviceManager;
         this.setActive = setActive;
         this.waitForSolver = waitForSolver;
         this.audioManager = audioManager;
-        this.onDisconnected = onDisconnected;
         normalMode = new NormalMode(audioManager, audioFocusManagerCall);
         mode = new BluetoothMode(audioManager, audioFocusManagerCall);
         wrapper.apply(BluetoothDevice.this::setPlatformConnectionState);
     }
 
     @NonNull
     @Override
-    protected Promise<Boolean> connect() {
+    protected Promise<Boolean> connect(@NonNull LastConnectionStateType lastConnectionStateType) {
```

Comment:
I didn't make the connect & disconnect method to have the connection type to prevent possibility for developers to actually choose what would be the type (protected, meaning that package viability is still possible). I'm open to any counter argument as well here :p

Reply:
When we detect that BT SCO is DISCONNECTED we have to go through whole disconnect path in order to correctly update ConnectScheduler state which hold current device and to correctly release Bluetooth SCO, audio focus, mode in BluetoothDevice class.  You can check that part of code in my Pull request https://github.com/voxeet/voxeet-sdk-android-audio/pull/7.